addons/account/models/res_config_settings.py

```diff
@@ -163,6 +174,13 @@ def _onchange_tax_exigibility(self):
             }
         return res
 
+    @api.depends('company_id', 'terms_type', 'invoice_terms_html')
+    def _compute_terms_preview(self):
+        if self.env.company.terms_type == 'html' and self.terms_type == 'html':
+            self.preview_ready = True
+        else:
+            self.preview_ready = False
+
```

Comment:
What is the purpose of this function? It's not better to directly put an attribute invisible in the view with `terms_type == "html"`?

Reply:
@jbm-odoo the aim is to display the button if the terms_type == 'html' (radio option checked) and if they are already set on the company. This is to avoid having a preview button that leads to a 404 when the option was not saved on the company. See https://github.com/odoo/odoo/blob/de32a5aad56f07c7ba07fe9ec560c45f72adb4f6/addons/account/controllers/terms.py#L18